src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -160,12 +162,19 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
         status = new BitbucketBuildStatus(hash, statusDescription, state, url, key, name);
         new BitbucketChangesetCommentNotifier(bitbucket).buildStatus(status);
         if (result != null) {
-            listener.getLogger().println("[Bitbucket] Build result notified");
+            listener.getLogger().println("[Bitbucket] Build result notified: " + status.getState());
         }
     }
 
     private static @CheckForNull BitbucketSCMSource findBitbucketSCMSource(Run<?, ?> build) {
         SCMSource s = SCMSource.SourceByItem.findSource(build.getParent());
+        if (s instanceof NullSCMSource) {
+            // for instance PR merged on Bitbucket since the build has been started
+            ItemGroup<?> grandFather = build.getParent().getParent();
+            if (grandFather instanceof SCMSourceOwner) {
+                s = ((SCMSourceOwner) grandFather).getSCMSources().get(0);
```

Comment:
If a Jenkins multibranch job has multiple SCM sources (e.g. an original repository and a fork with distinct branch names), I think this might not select the correct one.

Reply:
On the other hand, because this plugin is using the older REST API that does not distinguish between repositories, it will work anyway, as long as the repositories are on the same server.

Reply:
But a similar fix might not be valid for <https://github.com/jenkinsci/github-checks-plugin/issues/196> because the GitHub API [Create a check run](https://docs.github.com/en/rest/reference/checks#create-a-check-run) is at POST /repos/{owner}/{repo}/check-runs and presumably has to be told the correct repository.

Reply:
This fix worked for me and was quite easy to implement. A solution to be really safe would be probably to save the url to which the INPROGRESS event has been sent to reuse it at build completion.

Reply:
Not only the URL but also the credential ID.

I think this change is fine for bitbucket-branch-source-plugin, but we need something more reliable for the equivalent issue in github-checks-plugin and in the future bitbucket-server-checks-plugin.